plugin/builtin/helmchartinflationgenerator/HelmChartInflationGenerator.go

```diff
@@ -98,6 +98,11 @@ func (p *plugin) validateArgs() (err error) {
 	// disabled).
 	if p.ValuesFile == "" {
 		p.ValuesFile = filepath.Join(p.ChartHome, p.Name, "values.yaml")
+
+		// If the version is specified, use the versioned values file.
+		if p.Version != "" {
+			p.ValuesFile = filepath.Join(p.ChartHome, fmt.Sprintf("%s-%s", p.Name, p.Version), p.Name, "values.yaml")
```

Comment:
nit: Could you convert lines 100-104 to an if/else? I think it will be easier to read that way

Reply:
https://github.com/kubernetes-sigs/kustomize/pull/4999/commits/380abb9e79ace1b03d15985e885e4b9e42c78ac7